### PR TITLE
Map java.util.UUID to ID instead of String

### PR DESCRIPTION
**Pull request by mriceron**
The scope of the PR is to map `java.util.UUID` to `ID` instead of `String`.

I'm not sure which is the best place to add a test for this since I just took a quick look at the project, I'm open to suggestions.

### CODE REVIEW

**Comment by ghostdogpr** [comment]
Awesome!

There was a test in `ExecutionSpec` and it still passes since the result is the same as before. If you want to check the returned type explicitly, you could add one here: https://github.com/ghostdogpr/caliban/blob/master/core/src/test/scala/caliban/schema/SchemaSpec.scala

File: core/src/test/scala/caliban/schema/SchemaSpec.scala [hunk 1]
```diff
@@ -1,10 +1,12 @@
 package caliban.schema
 
+import java.util.UUID
+
 import scala.concurrent.Future
 import caliban.introspection.adt.{ __DeprecatedArgs, __Type, __TypeKind }
 import caliban.schema.SchemaSpecUtils._
 import zio.test.Assertion._
-import zio.test._
+import zio.test.{ test, _ }
```
Comment (ghostdogpr) [inline core/src/test/scala/caliban/schema/SchemaSpec.scala]: This one isn't needed, is it?

Comment (mriceron) [reply core/src/test/scala/caliban/schema/SchemaSpec.scala]: Thanks, didn't notice it 👍

**Comment by ghostdogpr** [comment]
Thanks! I’ll merge it tomorrow before releasing a new version.

**Comment by mriceron** [comment]
Thanks to you for the library and the support! Looking forward to the release :D

**Pull request merged**